diff -urN calamares-3.1.11/CMakeLists.txt calamares-3.1.11-xdg-su/CMakeLists.txt
--- calamares-3.1.11/CMakeLists.txt	2017-12-20 10:22:06.000000000 +0100
+++ calamares-3.1.11-xdg-su/CMakeLists.txt	2017-12-31 13:36:38.812399804 +0100
@@ -112,7 +112,6 @@
 
 find_package( Qt5 ${QT_VERSION} CONFIG REQUIRED Core Gui Widgets LinguistTools Svg Quick QuickWidgets )
 find_package( YAMLCPP 0.5.1 REQUIRED )
-find_package( PolkitQt5-1 REQUIRED )
 
 # Find ECM once, and add it to the module search path; Calamares
 # modules that need ECM can do
@@ -341,13 +340,6 @@
 
 install(
   FILES
-      com.github.calamares.calamares.policy
-  DESTINATION
-      "${POLKITQT-1_POLICY_FILES_INSTALL_DIR}"
-)
-
-install(
-  FILES
       calamares.desktop
   DESTINATION
       ${CMAKE_INSTALL_DATADIR}/applications
diff -urN calamares-3.1.11/calamares.desktop calamares-3.1.11-xdg-su/calamares.desktop
--- calamares-3.1.11/calamares.desktop	2017-12-20 10:22:06.000000000 +0100
+++ calamares-3.1.11-xdg-su/calamares.desktop	2017-12-31 13:36:03.923903596 +0100
@@ -5,7 +5,7 @@
 GenericName=System Installer
 Keywords=calamares;system;installer
 TryExec=calamares
-Exec=pkexec /usr/bin/calamares
+Exec=xdg-su -c /usr/bin/calamares
 Comment=Calamares — System Installer
 Icon=calamares
 Terminal=false
diff -urN calamares-3.1.11/com.github.calamares.calamares.policy calamares-3.1.11-xdg-su/com.github.calamares.calamares.policy
--- calamares-3.1.11/com.github.calamares.calamares.policy	2017-12-20 10:22:06.000000000 +0100
+++ calamares-3.1.11-xdg-su/com.github.calamares.calamares.policy	1970-01-01 01:00:00.000000000 +0100
@@ -1,22 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE policyconfig PUBLIC
-"-//freedesktop//DTD PolicyKit Policy Configuration 1.0//EN"
-"http://www.freedesktop.org/standards/PolicyKit/1/policyconfig.dtd">
-<policyconfig>
-
- <vendor>Calamares</vendor>
- <vendor_url>https://github.com/calamares</vendor_url>
-
- <action id="com.github.calamares.calamares.pkexec.run">
-    <description>Run Installer</description>
-    <message>Authentication is required to run the installation program</message>
-    <icon_name>drive-harddisk</icon_name>
-    <defaults>
-     <allow_any>no</allow_any>
-     <allow_inactive>no</allow_inactive>
-     <allow_active>auth_admin</allow_active>
-    </defaults>
-    <annotate key="org.freedesktop.policykit.exec.path">/usr/bin/calamares</annotate>
-    <annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
- </action>
-</policyconfig>
\ No newline at end of file
